src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -37,17 +39,37 @@ def are_broadcast_compatible(left: TypeInfo, right: TypeInfo) -> bool:
     >>> are_broadcast_compatible(int_field_t, int_scalar_t)
     True
 
+    >>> from functional.iterator.runtime import CartesianAxis
+    >>> Edge = CartesianAxis("Edge")
+    >>> K = CartesianAxis("K")
+    >>> are_broadcast_compatible(
+    ...     TypeInfo(ct.FieldType(dtype=ct.ScalarType(kind=ct.ScalarKind.INT64), dims=[K])),
+    ...     TypeInfo(ct.FieldType(dtype=ct.ScalarType(kind=ct.ScalarKind.INT64), dims=[Edge, K])),
+    ... )
+    True
+
+    >>> are_broadcast_compatible(
+    ...     TypeInfo(ct.FieldType(dtype=ct.ScalarType(kind=ct.ScalarKind.INT64), dims=[Edge])),
+    ...     TypeInfo(ct.FieldType(dtype=ct.ScalarType(kind=ct.ScalarKind.INT64), dims=[Edge, K])),
+    ... )
+    True
     """
     both_dims_given = bool(left.dims and right.dims)
     both_dims_given &= not isinstance(left.dims, GenericDimensions)
     both_dims_given &= not isinstance(right.dims, GenericDimensions)
-    if both_dims_given and any(
-        ldim != rdim
-        for ldim, rdim in zip(
-            left.dims, right.dims  # type: ignore[arg-type]  # we know they are lists here
+    if (
+        both_dims_given and isinstance(left.dims, list) and isinstance(right.dims, list)
+    ):  # isinstance calls are for mypy
+        smaller_dims, bigger_dims = (
+            (left.dims, right.dims)
+            if len(left.dims) <= len(right.dims)
+            else (right.dims, left.dims)
         )
-    ):
-        return False
+        if smaller_dims[0] in bigger_dims and smaller_dims[-1] in bigger_dims:
```

Comment:
```suggestion
        if set(left.dims).issubset(right.dims) or set(right.dims).issubset(left.dims):
          return True
```
May I'm missing the point, but can't we just do something like this?

Reply:
If you can prove that the order is not important, I would love to. For example we'd be saying `[K, Edge]` can be implicitly broadcast to `[Edge, K]`.

Reply:
With respect to iterator IR this is no problem. I tested this:
```
@field_operator
def add(inp1: Field[[IDim, JDim], float64], inp2: Field[[JDim, IDim], float64]) -> Field[[IDim, JDim], float64]:
    return inp1+inp2
```
and it does work as expected as long you give the right output type. The problem is that there is no good/easy way to decide what the output type should be. Is it `Field[[IDim, JDim], float64]` or `Field[[IDim, JDim], float64]`. Right now `inp1+inp2` will give you `Field[[IDim, JDim], float64]`, but `inp1(Ioff[1])+inp2(Joff[1])` gives `Field[[JDim, IDim], float64]`. I think it is not worth making the frontend capable of this, so let's leave it as is.